x509/bash.go

```diff
@@ -0,0 +1,50 @@
+//go:build !windows
+
+package x509
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/foundriesio/fioctl/subcommands"
+)
+
+func run(name string, arg ...string) string {
+	cmd := exec.Command(name, arg...)
+	cmd.Stderr = os.Stderr
+	out, err := cmd.Output()
+	subcommands.DieNotNil(err)
+	return string(out)
+}
+
+func CreateFactoryCa(ou string) string {
+	run("./create_ca")
+	return string(readFile(FactoryCaCertFile))
+}
+
+func CreateDeviceCa(cn string, ou string) string {
+	run("./create_device_ca", DeviceCaKeyFile, DeviceCaCertFile)
+	return string(readFile(DeviceCaCertFile))
+}
+
+func SignTlsCsr(csrPemBytes []byte) string {
+	return run("./sign_tls_csr", TlsCsrFile)
+}
+
+func SignCaCsr(csrPemBytes []byte, csrFileName string) string {
+	// If csrFileName contains a "*", csrPemBytes is written to a temp file to be used inside the script
+	var fileName string
+	if strings.Contains(csrFileName, "*") {
```

Comment:
I've spent a few minutes to understand _WHY_ this is special cased like this.

Although this technically works, I see a few problems with it:
1. The `x509/bash.go:SignCaCsr` is _polluted_ with the temp file handling, only specific to the El2Go.
2. The `x509/golang.go:SignCaCsr` is _polluted_ with the `csrFileName` it doesn't use at all.

So, I'd choose which of the above problems needs to be solved, and eliminate it.

To me, it seems that the second problem has a higher priority, as we're looking forward to switch to the `golang` implementation in the future.
So, it looks like we need to do smth like this:
- Define the `func SignCaCsr(csrPemBytes []byte) string` to only accept bytes.
- This works as is for the `ca_create` use case.
- For the `el2go` use case we need to incorporate the temp file logic somehow.
  I see two options how to solve this:
  * A: implement the temp file handling in el2go via `if runtime.OS == "windows"`.
  * B (my preference): define a separate function for el2go like `SignEl2GoCsr` which would:
    + hide the temp file specifics in `bash.go` file;
    + just call the SignCaCsr in golang.go.

Reply:
Having a separate `SignEl2GoCsr` makes sense. Implementing it now.